nc/nc.go

```diff
@@ -474,3 +476,36 @@ func (c *Client) Stop() error {
 
 	return nil
 }
+
+// Revoke revokes the NC communication token
+func (c *Client) Revoke(purgeSession bool) bool {
+	c.startMu.Lock()
+	defer c.startMu.Unlock()
+
+	if c.started {
```

Comment:
Technically management goroutine could still be running at this point, because `Stop` just calls `cancelConnecting` and doesn't wait for any kind of confirmation. So at this point management loop could be attempting to fetch new creds, connecting to NC, etc. Previously it was not an issue because in any context, there would always be a big delay between Starting/Stopping the NC. I'm not sure if now it's completely safe, since `Revoke` is called right after `Stop`. Might not be a big deal, since it's just the dev

Reply:
I checked what will happen when credentials will be revoked without stopping the loop. The manager will just try to fetch new credentials. So it does not look like can break anything. In our code stop is esured to be called before revoke. And even if the management loop will not be able to finish after stop call before calling revoke, it will finish when trying to get new creds.